### PR TITLE
Move flowkit package to its own module

### PR DESCRIPTION
**Pull request by dsainati1**
This separates the flowkit package into its own module at `github.com/onflow/flow-cli/pkg/flowkit`. This way, the Cadence language server can depend on the flowkit without introducing a circular dependency between the CLI and the language server. 
______

For contributor use:

- [X] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,27 +1,166 @@
...
+	google.golang.org/grpc v1.45.0 // indirect
+	google.golang.org/protobuf v1.28.0 // indirect
+	gopkg.in/ini.v1 v1.66.2 // indirect
+	gopkg.in/warnings.v0 v0.1.2 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	lukechampine.com/blake3 v1.1.7 // indirect
+)
+
+replace github.com/onflow/flow-cli/pkg/flowkit => ./pkg/flowkit
```
Comment (turbolent) [inline go.mod:166]: This shouldn't be committed. I guess we need it temporarily until we can point to a commit?

**Review by turbolent — APPROVED** [review]
Modularizing flowkit would be great, as we can make use of it in various tools, e.g. in the future in the emulator for the debugger.

In addition to splitting it up inside of this repo, maybe we can also consider moving it to the Go SDK or a separate repo?

**Comment by turbolent** [comment]
@dsainati1 Please wait until @sideninja had a chance to look at it

File: go.mod (hunk 1 above)
Comment (dsainati1) [reply go.mod:166]: Yeah; until we have a commit where this package exists I think we need this for now.

**Comment by dsainati1** [comment]
> In addition to splitting it up inside of this repo, maybe we can also consider moving it to the Go SDK

We could consider moving this to another repo, but the SDK would not work, I think, because flowkit depends on the emulator. At the very least it needs to be higher up in the dependency chain than that.

**Comment by bjartek** [comment]
What about putting it in a repo of its own?

**Comment by bluesign** [comment]
agreed with @bjartek, I think it is better to separate into new repo too. 

In general it is a bit mess, especially fvm - flow-go , SDK - flow-go etc. I think need to think all ( not focused only here, and should make a re-org )

**Comment by dsainati1** [comment]
Agreed, but this PR both makes that easier achieve and unblocks me on something else I am working on. We can follow up in another PR to make this its own repo.

**Review by psiemens — COMMENTED** [review]
Gregor and I had talked about eventually moving flowkit to its own repository. I think this is a good first step!

Gregor's out until the 14th. @sukantoraymond and I can help merge this, but I'd like to wait until Wednesday as we are planning to cut a release today/tomorrow and would need a bit more time to test the upgrade to Go 1.18. @dsainati1 does that work for you?

**Comment by dsainati1** [comment]
Yeah I can wait until then no problem.

**Review by psiemens — APPROVED** [review]

**Comment by sideninja** [comment]
I would say that extracting flowkit more would be conditioned on making sure we refactor it in places according to this https://github.com/onflow/flow-cli/issues/493
It has been my plan to make these steps, but I feel we should first make sure it works like a well-behaved library.

**Pull request merged**